src/App.js

```diff
@@ -9,16 +14,28 @@ import './App.css';
 import List from './pages/List/List';
 import AddItem from './pages/AddItem/AddItem';
 import NotFound from './pages/NotFound/NotFound';
+import Home from './pages/Home/Home';
 
 // Components
 import NavMenu from './components/NavMenu/NavMenu';
 
 function App() {
+  const [haveToken, setHaveToken] = useState(false);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    if (token) {
+      setHaveToken(true);
+    }
+  }, []);
```

Comment:
So I think the easiest way to handle this in the merge is to simply use the listId state as the conditional on the places where we use a `<Redirect />`. Because we are originally checking for a token as the condition, and `listId `depends on the token existing to be resolved to `truthy`, I believe everything will work the same way, and we can also use the `listId` check to address some of your other concerns, as I believe you also suggested.

Reply:
For better state management, I recommend saving the token value to app state:
```javascript
const [userToken, setUserToken] = useState('');
```
This gives you more flexibility than saving a boolean of its existence. Because this state is at App level, all child components of App.js have access to it for reuse by passing it down as props or the useContext. Because of Javascript's loosy goosy variables, you can use:
```javascript
{userToken ? <Redirect to="/list" /> :  <Home saveToken={saveToken} />}
// or
Boolean(userToken)
//or 
!!userToken
```
The first option is very common. The second is very clear and explicit. The last could be confusing when you need to do `!userToken`.  Here's more info on [Boolean coercion](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean).

Reply:
Will do, this definitely makes more sense.